src/Libraries/Microsoft.Extensions.AI.AzureAIInference/AzureAIInferenceChatClient.cs

```diff
@@ -449,7 +449,7 @@ private IEnumerable<ChatRequestMessage> ToAzureAIInferenceChatMessages(IEnumerab
                              callRequest.CallId,
                              new FunctionCall(
                                  callRequest.Name,
-                                 JsonSerializer.Serialize(callRequest.Arguments, JsonContext.GetTypeInfo(typeof(IDictionary<string, object>), ToolCallJsonSerializerOptions)))));
+                                 JsonSerializer.Serialize(callRequest.Arguments, ToolCallJsonSerializerOptions.GetTypeInfo(typeof(IDictionary<string, object>))))));
```

Comment:
Similarly here. My intent was that if the supplied JSO didn't contain the relevant type, it'd fall back to using the built-in one. Why don't we want that?

Reply:
See https://github.com/dotnet/extensions/pull/5630#discussion_r1839703731. We should use `ToolCallJsonSerializerOptions` for any type containing `object`.